lib/utils.coffee

```diff
@@ -292,10 +293,12 @@ timeoutPromise = (ms, promise) ->
 # utils.requestAsync({ url: 'http://example.com' }).then (response) ->
 # 	console.log(response)
 ###
-exports.requestAsync = (options) ->
+exports.requestAsync = (options, retriesRemaining = undefined) ->
```

Comment:
Ok, that makes sense, I'll move it to the JSDoc.

To briefly dive on the TypeScript diversion (with maybe unwarranted enthusiasm): those two options aren't equivalent in TS, so it's not just style.

Firstly, because `x?: number` has the same type within the function as `x: number | undefined`, not `x: number | null`. `retriesRemaining: number | null = undefined` isn't valid (assuming strictNullChecks)

Secondly, because the `?` doesn't just change the type of the argument inside the function, it also makes the argument optional. With `x: number | undefined` you can explicitly pass undefined, but you can't not pass any argument at all. With `x?: number` you can pass undefined implicitly, and I think that's probably the idiomatic option.

[Playground example](https://www.typescriptlang.org/play/#src=function%20f(x%3F%3A%20string)%20%7B%20%20%20%20%0A%7D%0A%0Afunction%20g(x%3A%20string%20%7C%20undefined)%20%7B%0A%0A%7D%0A%0Af()%0Ag()%20%2F%2F%20%3C--%20different%20behaviour%0A%0Af(undefined)%0Ag(undefined)%0A%0Af(%22a%20string%22)%0Ag(%22a%20string%22)%0A%0A%2F%2F%20I%20haven't%20checked%2C%20but%20I'm%20fairly%20sure%20these%20two%20would%0A%2F%2F%20both%20fail%20if%20strictNullChecks%20was%20on%20here%3A%0Af(null)%0Ag(null))